website/project/views/tag.py

```diff
@@ -48,9 +48,37 @@ def project_add_tag(auth, node, **kwargs):
 @must_have_permission('write')
 @must_not_be_registration
 def project_remove_tag(auth, node, **kwargs):
-
     data = request.get_json()
     tag = data['tag']
     if tag:
         node.remove_tag(tag=tag, auth=auth)
         return {'status': 'success'}
+
+
+@must_be_valid_project  # injects project
+@must_have_permission('write')
+@must_not_be_registration
+def file_add_tag(auth, node, guid, **kwargs):
+    data = request.get_json()
+    tag = data['tag']
```

Comment:
If `tag` was not submitted as part of the `POST` request, this line will raise a `KeyError`. Use `dict.get` to access it.